### PR TITLE
Moved flagutil to separate package; expanded blockgen; blockgen dockerfile added.

### PR DESCRIPTION
**Pull request by bwplotka**
This is needed for remote read benchmarks as the part of #488

Signed-off-by: Bartek Plotka <bwplotka@gmail.com>

### CODE REVIEW

File: cmd/thanos/bucket.go [hunk 1]
```diff
@@ -10,6 +10,8 @@ import (
 	"text/template"
 	"time"
 
+	"github.com/improbable-eng/thanos/pkg/flagutil"
+
```
Comment (povilasv) [inline cmd/thanos/bucket.go]: remove the space please

File: cmd/thanos/compact.go [hunk 2]
```diff
@@ -12,6 +12,8 @@ import (
 	"strings"
 	"time"
 
+	"github.com/improbable-eng/thanos/pkg/flagutil"
+
```
Comment (povilasv) [inline cmd/thanos/compact.go]: ditto

File: cmd/thanos/receive.go [hunk 3]
```diff
@@ -7,6 +7,8 @@ import (
 	"sync"
 	"time"
 
+	"github.com/improbable-eng/thanos/pkg/flagutil"
+
```
Comment (povilasv) [inline cmd/thanos/receive.go]: ditto

File: cmd/thanos/store.go [hunk 4]
```diff
@@ -5,6 +5,8 @@ import (
 	"net"
 	"time"
 
+	"github.com/improbable-eng/thanos/pkg/flagutil"
+
```
Comment (povilasv) [inline cmd/thanos/store.go]: ditto

File: pkg/flagutil/pathorcontent.go [hunk 5]
```diff
@@ -0,0 +1,53 @@
+package flagutil
+
+import (
+	"github.com/pkg/errors"
+	"gopkg.in/alecthomas/kingpin.v2"
+	"io/ioutil"
+)
+
+type PathOrContent struct {
```
Comment (povilasv) [inline pkg/flagutil/pathorcontent.go]: add a comment on top of this

File: tutorials/kubernetes-demo/blockgen/main.go [hunk 6]
```diff
@@ -2,13 +2,15 @@ package main
...
-	"io/ioutil"
 	"math/rand"
 	"os"
 	"path/filepath"
+	"strconv"
 	"strings"
 	"time"
 
+	"github.com/improbable-eng/thanos/pkg/flagutil"
+
```
Comment (povilasv) [inline tutorials/kubernetes-demo/blockgen/main.go]: ditto

**Comment by bwplotka** [comment]
We need proper linter for those empty lines.... goimports is not helpful ):

File: tutorials/kubernetes-demo/blockgen/main.go [hunk 7]
```diff
@@ -300,3 +359,81 @@ type gen interface {
 	Ts() int64
 	Value() float64
 }
+
+//
+//func genTSDBBlock(
```
Comment (brancz) [inline tutorials/kubernetes-demo/blockgen/main.go]: if you intend to leave this here I'd suggest we put at least a comment there explaining why it's here and why it's commented out

Comment (bwplotka) [reply tutorials/kubernetes-demo/blockgen/main.go]: I am experimenting a lot on this branch, so will close this PR for now to avoid confusion

**Comment by bwplotka** [comment]
I am experimenting a lot on this branch, so will close this PR for now to avoid confusion

**Pull request closed without merging**